src/compiler/types/validate-build-package-json.ts

```diff
@@ -27,22 +19,20 @@ export const validateBuildPackageJson = async (
   compilerCtx: d.CompilerCtx,
   buildCtx: d.BuildCtx
 ): Promise<void> => {
-  if (config.watch) {
-    return;
-  }
-  if (buildCtx.packageJson == null) {
+  if (config.watch || buildCtx.packageJson == null) {
     return;
   }
 
+  // Validate any output target that the user has designated as the "primary"
+  // target that is bundled with their distribution
+  validatePrimaryPackageOutputTarget(config, compilerCtx, buildCtx);
+
   const distCollectionOutputTargets = config.outputTargets.filter(isOutputTargetDistCollection);
-  const typesOutputTargets = config.outputTargets.filter(isOutputTargetDistTypes);
-  await Promise.all([
-    ...distCollectionOutputTargets.map((distCollectionOT) =>
+  await Promise.all(
+    distCollectionOutputTargets.map((distCollectionOT) =>
```

Comment:
There was a spread before because we also `await`ed the `validateTypes` and `validateModule` methods. Since that validation is now based on the "primary" target, I removed those calls so we only needed to `await` the validation for each of the other fields that may exist on `dist-collection` here now